File: android/src/main/kotlin/dev/jerrykhw/flutter_kakao_maps_sdk/model/CompassOptions.kt
package dev.jerrykhw.flutter_kakao_maps_sdk.model

import com.kakao.vectormap.PoiScale
import org.json.JSONObject

data class CompassOptions(
    var enabled: Boolean,
    var position: KakaoMapPosition,
)

fun JSONObject.toCompassOptions(): CompassOptions {
    return CompassOptions(
        this.getBoolean("enabled"),
        this.getJSONObject("position").toKakaoMapPosition(),
    )
}

File: lib/src/model/compass_options.dart
part of flutter_kakao_maps_sdk;

class CompassOptions {
  final bool enabled;
  final KakaoMapPosition position;

  const CompassOptions({
    this.enabled = true,
    required this.position,
  });

  CompassOptions copyWith({
    bool? enabled,
    KakaoMapPosition? position,
  }) =>
      CompassOptions(
        enabled: enabled ?? this.enabled,
        position: position ?? this.position,
      );
}

extension CompassOptionsExtension on CompassOptions {
  Map<String, dynamic> toMap() {
    return {
      "enabled": enabled,
      "position": position.toMap(),
    };
  }
}
